src/main/java/net/imagej/ops/filter/derivative/DirectionalDerivativeRAI.java

```diff
@@ -0,0 +1,97 @@
+package net.imagej.ops.filter.derivative;
+
+import net.imagej.ops.Ops;
+import net.imagej.ops.Ops.Filter.DirectionalDerivative;
+import net.imagej.ops.special.chain.RAIs;
+import net.imagej.ops.special.computer.BinaryComputerOp;
+import net.imagej.ops.special.computer.UnaryComputerOp;
+import net.imagej.ops.special.function.UnaryFunctionOp;
+import net.imagej.ops.special.hybrid.AbstractUnaryHybridCF;
+import net.imglib2.RandomAccessibleInterval;
+import net.imglib2.type.numeric.RealType;
+import net.imglib2.view.IntervalView;
+import net.imglib2.view.Views;
+
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+
+/**
+ * Calculates the derivative (with sobel kernel) of an image in a given dimension.
+ * 
+ * @author Eike Heinz, University of Konstanz
+ *
+ * @param <T> type of input
+ */
+@Plugin(type = Ops.Filter.DirectionalDerivative.class, name = Ops.Filter.DirectionalDerivative.NAME)
+public class DirectionalDerivativeRAI<T extends RealType<T>>
+		extends AbstractUnaryHybridCF<RandomAccessibleInterval<T>, RandomAccessibleInterval<T>>
+		implements DirectionalDerivative {
+
+	@Parameter
+	private int dimension;
+
+	private UnaryComputerOp<RandomAccessibleInterval<T>, RandomAccessibleInterval<T>> copyRAI;
+
+	private UnaryFunctionOp<RandomAccessibleInterval<T>, RandomAccessibleInterval<T>> createRAIFromRAI;
+
+	private UnaryComputerOp<RandomAccessibleInterval<T>, RandomAccessibleInterval<T>> yConvolverRotated;
+
+	private UnaryComputerOp<RandomAccessibleInterval<T>, RandomAccessibleInterval<T>> xConvolverRotated;
+
+	private BinaryComputerOp<RandomAccessibleInterval<T>, RandomAccessibleInterval<T>, RandomAccessibleInterval<T>> addOp;
+
+	private UnaryComputerOp<RandomAccessibleInterval<T>, RandomAccessibleInterval<T>> yConvolver;
+
+	private UnaryComputerOp<RandomAccessibleInterval<T>, RandomAccessibleInterval<T>> xConvolver;
+
+	@Override
+	public void initialize() {
+		RandomAccessibleInterval<T> kernel = ops().create().kernelSobelSeparated();
+		IntervalView<T> kernelX = Views.hyperSlice(Views.hyperSlice(kernel, 3, 0), 2, 0);
+		IntervalView<T> kernelY = Views.hyperSlice(Views.hyperSlice(kernel, 3, 0), 2, 1);
+
+		addOp = RAIs.binaryComputer(ops(), Ops.Math.Add.class, in(), in());
+
+		yConvolver = RAIs.computer(ops(), Ops.Filter.Convolve.class, in(), kernelY);
+		xConvolver = RAIs.computer(ops(), Ops.Filter.Convolve.class, in(), kernelX);
+
+		copyRAI = RAIs.computer(ops(), Ops.Copy.RAI.class, in());
+		createRAIFromRAI = RAIs.function(ops(), Ops.Create.Img.class, in());
+
+		IntervalView<T> filter = Views.rotate(kernelY, 0, 1);
```

Comment:
no need to create variable here. you can put `Views.rotate(..)` directly in the ops call.